spec/validators/claim/base_claim_validator_spec.rb

```diff
@@ -25,69 +25,196 @@
+        end
+
+        # TODO: not working
+        xit 'validation is performed on aggregated values' do
+          # allow(invalid_claim).to receive(:total).and_return(0.0)
+          invalid_claim.update_attribute(:total, 0.0)
+          expect { invalid_claim.submit! }.to raise_error StateMachines::InvalidTransition, /reason.*total.*/i
+        end
+      end
 
-        before { claim_with_nil_values.force_validation=false }
+      context 'when saving as draft' do
+        context '...and validation is forced' do
+          before { invalid_claim.force_validation = true }
 
-        context 'and it is coming from the api' do
-          before { claim_with_nil_values.source = 'api' }
-          it 'should validate presence of case_type, court, case_number, advocate_category, offence' do
-            expect(claim_with_nil_values).to_not be_valid
+          it 'validation is performed' do
+            expect(invalid_claim).to_not be_valid
           end
         end
 
-        context 'and it is coming from the web app' do
-          before { claim_with_nil_values.source = 'web' }
-          it 'should NOT validate presence of case_type, court, case_number, advocate_category, offence' do
-            expect(claim_with_nil_values).to be_valid
+        context '...and validation is NOT forced' do
+          before { invalid_claim.force_validation = false }
+
+          context '...and it is coming from the api' do
+            before { invalid_claim.source = 'api' }
+
+            it 'validation is performed' do
+              expect(invalid_claim).to_not be_valid
+            end
+          end
+
+          context '...and it is coming from the web app' do
+            before { invalid_claim.source = 'web' }
+
+            it 'validation is NOT performed' do
+              expect(invalid_claim).to be_valid
+            end
           end
         end
+      end
+    end
 
+    context 'when claim is in archived_pending_delete state' do
+      let(:claim) { create(:archived_pending_delete_claim) }
+
+      before do
+        nulify_fields_on_record(claim, :case_type, :court, :case_number, :advocate_category, :offence, :estimated_trial_length, :actual_trial_length)
+        claim.force_validation = false
+      end
+
+      it 'validation is NOT performed' do
+        expect(claim).to be_valid
       end
     end
 
-    context 'when claim is NOT draft' do
+    context 'when claim is in submitted state' do
+      before do
+        claim.submit!
+        claim.force_validation = false
+        nulify_fields_on_record(claim, :case_type, :court, :case_number, :advocate_category, :offence, :estimated_trial_length, :actual_trial_length)
+        claim.defendants.destroy_all
+        claim.fees.destroy_all
+        claim.expenses.destroy_all
+      end
 
-      before(:each) do
-        claim.force_validation=false
+      it 'validation is performed' do
```

Comment:
should probably be `it 'claim is not valid`

Reply:
I think it's okay in the context of `Validation is performed here` versus not being performed later